simpleviewinc/keg-test-consumer
# Fix demo app

**Ticket**: [ZEN-456](https://jira.simpleviewtools.com/browse/ZEN-456)

[Semver-Status] patch

> ### Notes
> * This pr is dependent on two other PR's
>   * [tap-events-force](https://github.com/simpleviewinc/tap-events-force/pull/99)
>   * [keg-hub](https://github.com/simpleviewinc/keg-hub/pull/88)
> * Those PRs should be reviewed along with this one

## Context

* The consumer was throwing errors when running with the latest build of the Sessions Component

## Goal

* Fix the errors so the consumer runs properly with the Sessions Component
* Clean up a few items to make testing easier

## Updates

* `.gitignore`
  * Added sessionsBuild folder
* `container/docker-compose.yml`
  * Included `KEG_NODE_VERSION` when running compose build command
* `container/values.yml`
  * Added the default `KEG_IMG_FROM` env
  * .Will be used later when the image from updates are complete
* `package.json`
  * Added a number of helper scripts and dependencies
*. `container/Dockerfile`
  * Updated to use the correct ENV, and port used in other locations
  * Removed other exposed ports no longer needed
* `container/mutagen.yml`
  * Added sync action for connecting to the tap folder
* `scripts/dockerConsumerBuild.sh` && `scripts/dockerEVFBuild.sh`
  * Helper scripts to build inside the docker container, then copy the artifact back to the host machine
  * These scripts require the changes in [this PR](https://github.com/simpleviewinc/keg-hub/pull/88) to run properly
    * Be sure to switch to that branch of keg-hub when running this PR's tests

## Testing

**Setup**
* Switch to the `add-tap-docker-tasks` branch for Keg-Hub, if those changes have not already been merged into develop.
  * Run command => `keg cli && keg pr 88`
* Switch to the `fix-rollup-export` branch for `tap-events-force` if those changes have not already been merged into develop.
  * Run command => `keg evf && keg pr 99`
* Pull this PR => `keg consumer && keg pr 10`

**Test 1**
* Run the consumer package => `keg consumer pack run docker.pkg.github.com/simpleviewinc/keg-packages/keg-test-consumer:fix-demo-app`
  * This package is using Session Component build from the changes of [this PR](https://github.com/simpleviewinc/tap-events-force/pull/99)
  * Ensure everything works as expected

**Test 2**
* Start the consumer => `keg consumer start`
  * Once it boots, stop the app `cntl+c`
* Sync the event-force tap => `keg consumer sync evf:install`
* Navigate to the consumer tap folder and run the sessions build script
  * `keg consumer && yarn docker:build:evf`
  * This should connect to the consumer container, and build the events force sessions component
  * It then copies the build from the container to the host machine at `<consumer-root-folder>/sessionsBuild`
  * Ensure the folder was copied to your host machine
  * Inspect the folder files, and ensure everything looks correct

**Test 3**
* Run the command `yarn docker:build:consumer`
  * This should connect to the consumer container, and build the tap-consumer app
  * It uses the `build:local` script from package.json to ensure if can be run locally

    >   *  **IMPORTANT** This build should not be used when pushing to github pages
    >   * The paths to assets is different, so it would not work properly

  * It then copies the build from the container to the host machine at `<consumer-root-folder>/build`
* After the build was copied to the host machine, run the command `yarn run:local`
  * This will run the **production** build of the consumer and the sessions component on you host machine
  * Navigate to http://localhost:5000 in your browser
  * Ensure everything works as expected
  * This test runs the production build of the app which is the same as what is used on github-pages
